### ISSUE TITLE
MySQL error on non-existent channelId

### ISSUE THREAD

**Issue by vpecinka** [mattermost/mattermost-server#9727]
#### Summary

When trying to view members of non-existent channelId the server report MySQL error due to malformed query.

#### Steps to reproduce
Usig Mattermost 5.4.0, the step are easy (id obfuscated)

```
curl -v -H 'Content-type: application/json' -d '{"channel_id": "1omxxxxxxxxfxjd9xh"}' -H 'Authorization: BEARER tce5xxxxxx7e' https://SITE_URL/api/v4/channels/members/34xxxxxxna/view
```
The API call returns 
```
{"id":"store.sql_channel.update_last_viewed_at.app_error","message":"Unable to update the last viewed at time","detailed_error":"","request_id":"pbf3ouhpgbb9zp4wuuht76ug3y","status_code":500}
```

#### Expected behavior
Should return something like 404 Not found

#### Observed behavior (that appears unintentional) 

The server logs

```
{"level":"error","ts":1540279611.0896354,"caller":"web/context.go:60","msg":"Unable to update the last viewed at time","path":"/api/v4/channels/members/34xxxxxxxxxpna/view","request_id":"utsa3fe3afdx7c8kfncprgdocy","ip_addr":"10.249.136.57","user_id":"34fxxxxxxxxxxxpna","method":"POST","err_where":"SqlChannelStore.UpdateLastViewedAt","http_code":500,"err_details":"channel_ids=j3poxxxxxxxxxc5ir, user_id=34xxxxxxxxxxxpdpna, Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'END,\n\t\t\t    LastViewedAt = CASE ChannelId  END,\n\t\t\t    LastUpdateAt = CASE Chann' at line 5"}
```

#### Possible fixes
I think it is somewhere here:  channel_store.go#L1590

**Comment by Charliekenney23** [comment]
I can take this issue on.

**Comment by amyblais** [comment]
Hi @vpecinka, thank you for reporting this!

@Charliekenney23 This is not marked as a Help Wanted Issue, but of course if you want to you can take a look at it and the contribution will count towards Hacktoberfest. Thank you!

**Comment by amyblais** [comment]
Hi @Charliekenney23, checking in on if you had a chance to take a look at this Issue? Feel free to also let me know if you didn't have a chance to look at it and I can also open a ticket for us to look at it internally.

**Comment by Charliekenney23** [comment]
@amyblais Sorry, I got caught up with something. I can definitely look into this tonight.

**Comment by amyblais** [comment]
@Charliekenney23 No worries!

**Comment by lieut-data** [comment]
I can reproduce this one! Thanks for spotting it, @vpecinka :)

**Comment by amyblais** [comment]
Created a ticket here for tracking: https://mattermost.atlassian.net/browse/MM-13073.

**Comment by amyblais** [comment]
Closing this as the Jira ticket has been resolved. The fix will be available in v5.8 release.
